### PR TITLE
AlibabaCloud LogService Exporter - Add logs exporter

### PR DESCRIPTION
**Pull request by shabicheng**
**Description:** 

add logs exporter for AlibabaCloud LogService Exporter

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** 

Unit tests added for logs exporter. Now code coverage is 99%.

Manually tested, sending to a LogService's instance.

**Documentation:** <Describe the documentation added.>

No need to update readme.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@shabicheng do you have a co-worker who can help review this PR and possibly become a co-codeowneer for the component?

File: exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	slsLogTimeUnixNano   = "timeUnixNano"
+	slsLogSeverityNumber = "severityNumber"
+	slsLogSeverityText   = "severityText"
+	slsLogName           = "name"
+	slsLogContent        = "content"
+	slsLogAttribute      = "attribute"
+	slsLogFlags          = "flags"
+)
+
+func logDataToLogService(logger *zap.Logger, ld pdata.Logs) ([]*sls.Log, int) {
```
Comment (tigrannajaryan) [inline exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go:38]: Looks like this translation function ignores the Resource part of the log data. Is this intentional?

File: exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+func mapLogRecordToLogService(lr pdata.LogRecord, logger *zap.Logger) *sls.Log {
+	if lr.Body().IsNil() {
+		return nil
+	}
+	var slsLog sls.Log
+
+	// pre alloc, refine if logContent's len > 16
+	slsLog.Contents = make([]*sls.LogContent, 0, 16)
```
Comment (tigrannajaryan) [inline exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go:81]: Nit: It is also possible to preallocate a slice of sls.LogContent and point to them from this slice.

File: exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+		// convert time nano to time seconds
+		slsLog.Time = proto.Uint32(uint32(lr.Timestamp() / 1000000000))
+	} else {
+		slsLog.Time = proto.Uint32(uint32(time.Now().Unix()))
+	}
+
+	return &slsLog
+}
+
+func convertAttributeValueToString(value pdata.AttributeValue, logger *zap.Logger) string {
```
Comment (tigrannajaryan) [inline exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go:143]: This appears very similar to https://github.com/open-telemetry/opentelemetry-collector/blob/8b67a09d745d9429a1d21f136d7f4283d56dd9be/translator/trace/protospan_translation.go#L114
Can you use that instead or there are differences?

File: exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go (hunk 1 above)
Comment (shabicheng) [reply exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go:46]: It is not used by SLS now, but we may need Resource and InstrumentationLibrary in the future. We have added it.

File: exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go (hunk 2 above)
Comment (shabicheng) [reply exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go]: done

File: exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go (hunk 3 above)
Comment (shabicheng) [reply exporter/alibabacloudlogserviceexporter/logsdata_to_logservice.go]: Great! `tracetranslator.AttributeValueToString` is what I want, Thanks a lot ~

**Comment by shabicheng** [comment]
> @shabicheng do you have a co-worker who can help review this PR and possibly become a co-codeowneer for the component?

Yeah, we have 4 guys working with observability, the exporter is reviewed inner before I push to github(We usually use gitlab in our company).

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
> > @shabicheng do you have a co-worker who can help review this PR and possibly become a co-codeowneer for the component?
> 
> Yeah, we have 4 guys working with observability, the exporter is reviewed inner before I push to github(We usually use gitlab in our company).

@shabicheng it would be great if you could do the reviews directly in this repository. We can add your co-workers as codeowners on this component and their review/approval will count.

**Pull request merged**